src/api/dependency-graph/src/graph-collector/graph.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * A Release represents a snapshot of a project.
+ *
+ * @param {string} version the version tag of this release
+ * @param {object} dependencies collection of regular dependencies with their versions
+ * @param {object} devDependencies collection of developer dependencies with their versions
+ * @param {object} peerDependencies collection of peer dependencies with their versions
+ */
+function Release(version, dependencies, devDependencies, peerDependencies) {
+  this.version = version;
+  this.dependencies = dependencies || {};
+  this.devDependencies = devDependencies || {};
+  this.peerDependencies = peerDependencies || {};
+}
+
+/**
+ * A Project represents an NPM package. A Project can be considered
+ * the dependency of another project depending on the dependency hierarchy.
+ *
+ * @param {string} name the name of the project, which may include the namespace
+ * @param {string} description the description of the project
+ * @param {string} repositoryUrl the repository URL to a GitHub remote repo
+ * @param {string} version the version tag
+ * @param {object} allDependencies collection of all dependencies
+ */
+function Project(name, description, repositoryUrl, version, allDependencies) {
+  this.name = name;
+  this.description = description;
+  this.repositoryUrl = repositoryUrl;
+  this.releases = {};
+
+  this.releases[version] = new Release(
```

Comment:
Do we care about historical releases?  Should we focus on the dependencies for the current version only, since you're going to snapshot the Telescope dependencies vs. try to keep them in sync?

Reply:
So, this code does not worry about historical releases of Telescope, it worries about all the dependencies that it is currently using, where some of these dependencies could feature different releases while still being the same project. For example, the `@babel/runtime` appears in the dependency tree as `7.16.7`, and again as `7.5.0`, but these releases belong to different projects in the tree.